### PR TITLE
CWMS-990 Added CumulusDss util object for building Dss path for a given Watershed/Product

### PR DESCRIPTION
**Pull request by rma-bryson**

### CODE REVIEW

File: cumulus-client/src/main/java/mil/army/usace/hec/cumulus/client/controllers/CumulusDssDataUtil.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    /**
+     * Builds Dss path string for given Watershed and Product.
+     * @param watershed - Watershed
+     * @param product - Product
+     * @return DSS path String
+     */
+    public static String buildDssPath(Watershed watershed, Product product) {
+        StringJoiner retVal = new StringJoiner("/");
+        String partA = "SHG";
```
Comment (rma-psmorris) [inline cumulus-client/src/main/java/mil/army/usace/hec/cumulus/client/controllers/CumulusDssDataUtil.java:24]: if any of these parts are blank or undefined in JSON, what do you get here? Wondering if you need null checks. empty pathname parts are fine too //

File: cumulus-client/src/main/java/mil/army/usace/hec/cumulus/client/controllers/CumulusDssDataUtil.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        String partB = watershed.getName();
+        String partC = product.getParameter();
+        String partD = DateTimeFormatter.ofPattern("ddMMMyyyy:HHmm").format(product.getAfter());
+        String partE = DateTimeFormatter.ofPattern("ddMMMyyyy:HHmm").format(product.getBefore());
+        String partF = product.getDssFPart();
+        retVal.add(partA).add(partB).add(partC).add(partD).add(partE).add(partF);
+        return retVal.toString();
```
Comment (rma-psmorris) [inline cumulus-client/src/main/java/mil/army/usace/hec/cumulus/client/controllers/CumulusDssDataUtil.java:31]: what does this string look like? it should be 
/A/B/C/D/E/F/ the D and E parts are optional

File: cumulus-client/src/test/java/mil/army/usace/hec/cumulus/client/controllers/TestCumulusDssData.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+final class TestCumulusDssData {
+
+    @Test
+    void testSerialization() throws IOException, URISyntaxException {
+        String watershedJson = readResourceFile("cumulus/json/watershed.json");
+        String productJson = readResourceFile("cumulus/json/product.json");
+        Watershed watershed = CumulusObjectMapper.mapJsonToObject(watershedJson, Watershed.class);
+        Product product = CumulusObjectMapper.mapJsonToObject(productJson, Product.class);
+        String dssPath = CumulusDssDataUtil.buildDssPath(watershed, product);
+        assertEquals("SHG/Cumberland Basin River/PRECIP/23Aug2021:0601/24Apr2022:1201/MBRFC-FORECAST", dssPath);
```
Comment (rma-psmorris) [inline cumulus-client/src/test/java/mil/army/usace/hec/cumulus/client/controllers/TestCumulusDssData.java]: I think those times are illegal and they dont look like they make sense. Probably better to leave them blank. Also, missing preceding and trailing "/"

**Review by rma-psmorris — CHANGES_REQUESTED** [review]

File: cumulus-client/src/test/java/mil/army/usace/hec/cumulus/client/controllers/TestCumulusDssData.java (hunk 3 above)
Comment (rma-bryson) [reply cumulus-client/src/test/java/mil/army/usace/hec/cumulus/client/controllers/TestCumulusDssData.java]: removed times and added preceding and trailing "/"

File: cumulus-client/src/main/java/mil/army/usace/hec/cumulus/client/controllers/CumulusDssDataUtil.java (hunk 2 above)
Comment (rma-bryson) [reply cumulus-client/src/main/java/mil/army/usace/hec/cumulus/client/controllers/CumulusDssDataUtil.java:31]: Added preceding and trailing "/"

File: cumulus-client/src/main/java/mil/army/usace/hec/cumulus/client/controllers/CumulusDssDataUtil.java (hunk 1 above)
Comment (rma-bryson) [reply cumulus-client/src/main/java/mil/army/usace/hec/cumulus/client/controllers/CumulusDssDataUtil.java:24]: Added null checks to turn into empty strings if parts are blank in JSON

**Review by rma-psmorris — APPROVED** [review]

**Pull request merged**